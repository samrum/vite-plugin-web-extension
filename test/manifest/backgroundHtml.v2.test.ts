import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("backgroundHtml");

runManifestV2Test("backgroundHtml", () => ({
  background: {
    page: `${resourceDir}/background.html`,
    persistent: false,
  },
}));
