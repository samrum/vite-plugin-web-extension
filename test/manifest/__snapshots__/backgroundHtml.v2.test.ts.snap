// Vitest Snapshot v1

exports[`backgroundHtml - Manifest V2 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log(\\"background\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/backgroundHtml/background.html",
    "fileName": "assets/test/manifest/resources/backgroundHtml/background.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/backgroundHtml/background.html": {
        "code": null,
        "originalLength": 229,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/backgroundHtml/background.js": {
        "code": "log(\\"background\\");",
        "originalLength": 55,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 18,
      },
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "test/manifest/resources/backgroundHtml/background",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "test/manifest/resources/backgroundHtml/background.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/backgroundHtml/background.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"background\\": {
    \\"page\\": \\"test/manifest/resources/backgroundHtml/background.html\\",
    \\"persistent\\": false
  }
}",
    "type": "asset",
  },
]
`;
